tensorboard/components/tf_wbr_string/tf-wbr-string.html

```diff
@@ -37,19 +37,21 @@
       properties: {
         /** The value to render with word breaks. */
         value: String,
+        /** The regular expression for specifying word break delimiters. */
```

Comment:
The code as written works only if `delimiterPattern` matches are always
exactly one character long (because of `idx + 1` in `_computeParts`).
Could we either (a) remove the footgun by interpreting this as a list of
characters rather than a full-fledged regex, (b) remove the footgun by
making the `_computeParts` code handle the general case, or (c) document
the footgun? or something else?

(If option (b) or (c), maybe clarify the docs to say “the string source
of a regular expression…” to avoid suggesting that it’s actually a
`RegExp` value.)

Reply:
I implemented (a) this morning but was unhappy with the result:
https://github.com/tensorflow/tensorboard/pull/3633/commits/5f1b06f82341ebe98ec48b8229ec38baa3f0c494

So I've gone with option (b) instead. I added some basic tests to validate.

Reply:
Okay, that’s fine with me, though it definitely increases the complexity
of the change.

FWIW, you probably want to avoid `new RegExp('[' + delimiters + ']');`,
which suffers from regex injection and thus doesn’t actually fulfill the
stated contract; instead, you might want [`_.escapeRegExp`][ld], since
the powers that be have decided in their infinite wisdom that
[`RegExp.escape` is too dangerous for us mere mortals to handle][tc].

[ld]: https://lodash.com/docs#escapeRegExp
[tc]: https://github.com/benjamingr/RegExp.escape/issues/37#issue-98309281

Reply:
I'm a little confused. I had changed things so that I was not wrapping the regex in '[' and ']'.  I think using _.escapeRegExp would mean callers couldn't specify patterns like "[a-g]" and get it to work as expected.